python/pyarrow/_flight.pyx

```diff
@@ -1320,6 +1399,11 @@ cdef class FlightClient(_Weakrefable):
             check_flight_status(CFlightClient.Connect(c_location, c_options
                                                       ).Value(&self.client))
 
+    def as_async(self) -> None:
+        if anyio is None:
+            raise RuntimeError("anyio is required for the async interface")
```

Comment:
I'm not sure why we would want `anyio`. Just start with plain `asyncio`?

Reply:
The main thing is to support both asyncio and Trio users (though I think anyio is unnecessary and we can just directly support them both)

Reply:
Why would we want that, though? asyncio is the de facto standard. And the anyio API for delegating stuff to external threads seems rather convoluted.

Reply:
I don't see a reason to just lock out a part of the ecosystem.

Reply:
And FWIW, one of the very first issues opened against ADBC was request for async support via anyio specifically, to support both Trio and asyncio